ai/core/UnityCatalog_AI_DevLoop.ipynb

```diff
@@ -0,0 +1,2233 @@
+    "print(\"Supermassive Black Hole (4 million M☉, like Sgr A*):\")\n",
+    "pprint(ast.literal_eval(smbh_result.value))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f8dff208-11bc-495e-9797-7bbd74eeac05",
+   "metadata": {},
+   "source": [
+    "## Sandbox Protection: Safeguarding Against Dangerous Executions\n",
+    "\n",
+    "One of the most powerful security enhancements in Unity Catalog AI 0.3.0 is the improved sandbox execution mode. This feature provides critical protection for production systems by isolating potentially hazardous function executions in separate processes.\n",
+    "\n",
+    "### Why Sandbox Protection Matters\n",
+    "\n",
+    "When working with AI agents that can call arbitrary functions, you face a fundamental challenge: users might unknowingly (or deliberately) provide parameters that could:\n",
+    "\n",
+    "1. **Crash your system** with unbounded computational demands\n",
+    "2. **Lock up resources** with infinite loops or excessive memory usage\n",
+    "3. **Access sensitive system resources** through imported modules\n",
+    "4. **Block other operations** by monopolizing computational resources\n",
+    "\n",
+    "The sandbox execution mode creates a protective barrier by running functions in isolated processes with strict resource limitations.\n",
+    "\n",
+    "### How Sandbox Protection Works\n",
+    "\n",
+    "In this section, we'll demonstrate how the sandbox works by creating a deliberately resource-intensive function: a Mandelbrot set calculator. While this function is perfectly safe with reasonable parameters, it can consume enormous computational resources when given large dimensions or iteration counts.\n",
+    "\n",
+    "The sandbox protection system:\n",
+    "\n",
+    "1. **Runs code in isolated processes** rather than in the main application process\n",
+    "2. **Enforces CPU time limits** to prevent excessive computation\n",
+    "3. **Restricts memory usage** to prevent memory exhaustion\n",
+    "4. **Blocks access to sensitive system modules** like `os`, `sys`, and `subprocess`\n",
+    "5. **Imposes execution timeouts** to ensure functions terminate\n",
+    "\n",
+    "For our demonstration, we've customized the timeout to just 3 seconds (compared to the default 30 seconds) to show how quickly the system can respond to potentially problematic executions.\n",
+    "\n",
+    "This protection is especially important for GenAI applications, where large language models might explore edge cases or extreme parameters that human users would typically avoid. By configuring the sandbox mode appropriately, you can ensure that even the most computationally demanding function calls will fail gracefully rather than bringing down your entire application."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "id": "5a0fe60e-7f71-41ca-be1b-27786f3a04d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def mandelbrot_set_calculator(\n",
```

Comment:
Should we move this code block to below `The Mandelbrot Example` section?

Reply:
I just realized that I completely forgot to write the markdown section for that code block. Adding it...